### DISCUSSION TITLE
MediaRecorder Misses Last Blob

### DISCUSSION THREAD

**Discussion by andrejhronco** [cables-gl/cables_docs#138] [! Welcome - readme first!]
When stopping recording, the last blob doesn't make it into the downloaded file (download gets called too soon, missing the last ~1sec of the recording). You can see this behavior if you log when `handleDataAvailable` gets called as well as when `download` gets called from `stopRecording`. If `download` is called on the `stop` event, the recordedBlobs length is correct.
`mediaRecorder.onstop = download`

**Comment by andrejhronco** [comment]
Let me know if this issue can be reproduced or if you need any more info. it's a simple fix that could prevent some confusion of why the last ~1sec is missing in people's videos.

**Comment by pandrr** [comment]
thank you for this fix!
it is implemented on dev.cables.gl and seems to work great
